task_processor/admin.py

```diff
@@ -18,3 +21,16 @@ def last_run_status(self, instance: RecurringTask) -> str | None:
         if last_run := instance.task_runs.order_by("-started_at").first():
             return last_run.result
         return None
+
+    def last_run_finished_at(self, instance: RecurringTask) -> datetime | None:
+        if last_run := instance.task_runs.order_by("-started_at").first():
+            return last_run.finished_at
+        return None
```

Comment:
What's the intention here? To add this to the list view, or just the detail view? At the moment, I'm not sure it's added to either? 🤔

Reply:
I'd also be reluctant to add this to the list display as it probably won't perform that well being run for all the records in the view.

Reply:
Good catch, added the field to `list_display`.

> I'd also be reluctant to add this to the list display as it probably won't perform that well being run for all the records in the view.

A similiar query is already run for all the records in the view — see `last_run_status`. I wouldn't worry too much about performance as the recurring task list is fairly constant.

Reply:
Yeah, fair enough. I guess there's an argument to say that we could cache the last run between the 2 methods here, but maybe that's over optimisation at this stage.